nodejs/week3/Mealsharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,202 @@
+const knex = require("../database");
+const express = require("express");
+const router = express.Router();
+//get method
+router.get("/meals", async (req, res) => {
+          try {
+                    const allMeals = await knex.from("meal").select("*");
+                    if (allMeals.length == 0) {
+                              res.status(404).send(
+                                        "Sorry data does not exists any more"
+                              );
+                    }
+                    res.send(allMeals);
+          } catch (err) {
+                    throw err;
+          }
+});
+//get method using id
+router.get("/meals/:id", async (req, res) => {
+          try {
+                    const mealId = req.params.id;
+                    const mealWithId = await knex
+                              .from("meal")
+                              .select("*")
+                              .where({ meal_id: mealId });
+                    if (mealWithId.length == 0) {
+                              res.status(404).send(
+                                        "Sorry data does not exists"
+                              );
+                    }
+                    res.send(mealWithId);
+          } catch (err) {
+                    throw err;
+          }
+});
+//post method
+router.post("/meals", async (req, res) => {
+          const query = {
+                    title: req.query.title,
+                    description: req.query.description,
+                    location: req.query.location,
+                    price: req.query.price,
+          };
+          if (query) {
+                    const qResponse = await knex("meal").insert(query);
+                    if (qResponse) {
+                              res.status(404).send("data added");
+                    }
+          }
+});
+
+//put method
+router.put("/meals", async (req, res) => {
+          const query = {
+                    title: req.query.title,
+                    description: req.query.description,
+                    location: req.query.location,
+                    price: req.query.price,
+          };
+          if (query) {
+                    const qResponse = await knex("meal")
+                              .where({ meal_id: req.query.id })
+                              .update(query);
+                    if (qResponse) {
+                              res.send("data updated");
+                    }
+          }
+});
+
+//Delete method
+router.delete("/meals/:id", async (req, res) => {
+          const mealsId = { meal_id: req.params.id };
+          if (mealsId) {
+                    const qResponse = await knex("meal").where(mealsId).del();
+                    if (qResponse) {
+                              res.status(404).send("Record deleted");
```

Comment:
what status code would you use to convey that deleting a meal was succesful?

404 is the status code for "Not found". All codes in the 4xx represent client errors. Does this case represent a client error?